src/modules/global_ban_protection/index.js

```diff
@@ -0,0 +1,61 @@
+const settings = require('../../settings');
+const twitch = require('../../utils/twitch');
+
+// Regular users are allowed 20 messages in 30 seconds, mods and higher
+// positions 100 messages in 30 seconds, see
+// https://discuss.dev.twitch.tv/t/message-limit-for-moderators/3042/4
+const regularMsgLimit = 20;
+const modMsgLimit = 100;
+const timeLimit = 30;
+
+class GlobalBanProtectionModule {
+    constructor() {
+        settings.add({
+            id: 'globalBanProtection',
+            name: 'Global Ban Protection',
+            description: 'Prevent sending a message if it would result in a global ban',
+            default: false,
+        });
+
+        this.msgLimit = -1;
+        this.msgTimeStamps = new Array();
+    }
+
+    onSendMessage(sendState) {
+        if (settings.get('globalBanProtection') === false) return;
+
+        if (this.msgLimit < 0) {
+            // Do this here for a reliable result, wasn't reliable in the constructor
+            if (twitch.getCurrentUserIsOwner() || twitch.getCurrentUserIsModerator()) {
+                this.msgLimit = modMsgLimit;
```

Comment:
instead of setting this on every send message, add a getter to this class for message limit.

```
get maxMessagesPerSecond() {
    return (twitch.getCurrentUserIsOwner() || twitch.getCurrentUserIsModerator()) ? MODERATOR_MAX_MESSAGES_PER_SECOND : MAX_MESSAGES_PER_SECOND;
}
```